ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/phase0/util/AttestationUtilPhase0.java

```diff
@@ -91,7 +79,19 @@ private boolean isFromFarFuture(
     final UInt64 discardAttestationsAfterMillis =
         currentTimeMillis.plus(
             secondsToMillis(MAX_FUTURE_SLOT_ALLOWANCE.times(specConfig.getSecondsPerSlot())));
-    return attestationSlotTimeMillis.isGreaterThan(discardAttestationsAfterMillis);
+    return attestationForkChoiceEligibleTimeMillis.isGreaterThan(discardAttestationsAfterMillis);
+  }
+
+  private boolean isCurrentTimeBeforeMinimumAttestationBroadcastTime(
+      final UInt64 attestationSlot, final UInt64 genesisTime, final UInt64 currentTimeMillis) {
+    final UInt64 minimumBroadcastTimeMillis =
+        minimumBroadcastTimeMillis(attestationSlot, genesisTime);
+    return currentTimeMillis.isLessThan(minimumBroadcastTimeMillis);
+  }
+
+  private boolean isCurrentTimeAfterAttestationPropagationSlotRange(
+      final UInt64 attestationSlot, final UInt64 genesisTime, final UInt64 currentTimeMillis) {
+    return maximumBroadcastTimeMillis(attestationSlot, genesisTime).isLessThan(currentTimeMillis);
```

Comment:
can we check code coverage, and if its not immaculate, then we should consider making tests for these - they could easily be package private static, and we could feed in values to validate with great coverage..

Reply:
Sure, I will extend current tests for Phase0. I didn't touch them but good idea to see we are covering all methods.